### ISSUE TITLE
Graduate Containers from Test Pilot to AMO

### ISSUE THREAD

**Issue by groovecoder** [mozilla/testpilot#2880]
* [x] Move the add-on and users to AMO
* [x] Write graduation post on Medium
* [ ] Update the testpilot.firefox.com page for Containers (Should I just send a PR to the containers.yml file?)

**Comment by johngruen** [comment]
@groovecoder can we put it on our medium instead? we want to start hosting grad reports off site.

**Comment by groovecoder** [comment]
Sure. Can I get access to that, or do you want to make the Copy Pasta™ from the doc?

**Comment by johngruen** [comment]
@groovecoder i can add you to our medium...We'll need elvin to review the report before we publish. I am making edits now.

**Comment by johngruen** [comment]
it is done

**Comment by groovecoder** [comment]
The post on medium is done and we'd like to post on Tuesday.

@wresuolc @johngruen - what are the remaining steps to updated the testpilot.firefox.com site for Tuesday?

**Comment by lmorchard** [comment]
FWIW, our next scheduled release of the site is the Tuesday after (10/17) and yesterday was the l10n string freeze for that release.

If we want to update the site on Tuesday (10/10), we *could* cherry pick a change to containers.yaml. But, if there are string changes, that's too short a lead time to get translations from contributors. Alternatively, we could live with untranslated en-US strings in the interim until the next scheduled release, assuming the contributions arrive.

(also fwiw, not trying to be jerky in mentioning all this - I don't think we've fully worked out or communicated a good strategy for all this yet)

**Comment by groovecoder** [comment]
Oh dang. ... did we have translations of the existing page? We would probably prefer to release sooner than later, because having the AMO listing AND the testpilot page is confusing people.

**Comment by lmorchard** [comment]
> Oh dang. ... did we have translations of the existing page? We would probably prefer to release sooner than later, because having the AMO listing AND the testpilot page is confusing people.

Not so much about translations of the existing page (which we should have) as much as translations of any new strings we need as part of the transition to graduated state. 

But, I forgot that since #2694, we switched to just linking to graduation content and use common strings that we should have localized already (or very soon). 

So, I don't think we have any strings that need to be changed for Containers. If [the Page Shot grad report](https://github.com/mozilla/testpilot/commit/5772a1d7c4b5a10ff0ab0abdfaae698da8c9fe0c) is a model (and I think it is), then we *should* be able to cherry pick something similar in without looping in l10n contributors.

**Comment by lmorchard** [comment]
And just as a note probably to myself, if I end up doing the cherry picking for this: It might be good to also grab a merge from the l10n branch, since there's been a lot of progress across locales in getting the strings from #2694 translated. https://pontoon.mozilla.org/projects/test-pilot-website/

**Comment by ghost** [comment]
So far we've had graduation reports on test pilot, not on blogs.  (we want them on blogs though).  So, short term, we could probably just make a graduation report for Containers that was a link to the blog, and longer term we could make that a smoother transition.

For that, it's just a matter of putting a `graduation_report` key in https://github.com/mozilla/testpilot/blob/master/content-src/experiments/containers.yaml with the link and any other explanatory text.  Medium links have trailing stuff in them, but maybe it's possible to get them ahead of time?  Otherwise we'd have to publish first and then update Test Pilot.

**Comment by lmorchard** [comment]
> So far we've had graduation reports on test pilot, not on blogs. (we want them on blogs though).

That changes with #2694 in the next prod push, though. We've removed support for `graduation_report` and added `graduation_url`. We *could* cherry-pick #2694 to expedite for Containers, but that might be trouble.

> For that, it's just a matter of putting a graduation_report key in https://github.com/mozilla/testpilot/blob/master/content-src/experiments/containers.yaml with the link and any other explanatory text. 

We could also add both `graduation_report` and `graduation_url`:

The `graduation_report` gives us short term text with a link - but with fallback en-US text in all locales because l10n won't have enough time.

Then, `graduation_url` will replace it with a simple link (with translations) after our next full prod push on 10/17

> Medium links have trailing stuff in them, but maybe it's possible to get them ahead of time? Otherwise we'd have to publish first and then update Test Pilot.

FWIW, [we already did that for Page Shot](https://github.com/mozilla/testpilot/commit/5772a1d7c4b5a10ff0ab0abdfaae698da8c9fe0c#diff-82065dcee5037cdccf8a6f186cf70635R98): Medium post is up but the update on Test Pilot isn't coming until the next prod push.

**Comment by groovecoder** [comment]
I tried locally on `master` branch to add both `graduation_report` and `graduation_url` but `graduation_report` seems to have no effect. In fact, when I looked at the old `tracking-protection.yaml` page, none of the content in `graduation_report` shows up there either.

Is there a plan (or code) to handle the old `yaml` files with `graduation_report`?

**Comment by fzzzy** [comment]
@groovecoder We had decided to not support graduation_report once graduation_url lands, but that won't be deployed until Tuesday the 17th. You could test that the graduation_report is working against the 2017-10-04 tag instead of master, and test graduation_url against master.

**Comment by groovecoder** [comment]
We'll just go with `graduation_url` then. I planned to publish the post on medium.com tomorrow anyway, so then we'll have the final url to add to `containers.yaml`.

I'll send a PR tomorrow for next week's deployment.

**Comment by lmorchard** [comment]
> We'll just go with graduation_url then. I planned to publish the post on medium.com tomorrow anyway, so then we'll have the final url to add to containers.yaml.

If you just add `graduation_url` in your PR and not also `graduation_report`, the update won't appear until next Tuesday (10/17).  So, if you're okay with waiting until next Tuesday for an update, that's cool. But it sounded like things were more urgent than that.

edit: deleted some details since fzzzy already covered them and I think I didn't quite understand

**Comment by groovecoder** [comment]
Yup, `graduation_url` on 10/17 is fine.

**Comment by ghost** [comment]
@groovecoder do you have a URL for this yet?  (so we can make the patch).  Our freeze is thursday.

**Comment by johngruen** [comment]
@groovecoder @wresuolc pr inbound with grad URL

**Comment by groovecoder** [comment]
@johngruen I win: https://github.com/mozilla/testpilot/pull/2923
